packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ip-ip-agoraOrder.controller.js

```diff
@@ -394,6 +411,53 @@ export default class AgoraIpOrderCtrl {
     return this.$state.go('^');
   }
 
+  getOfferDetails = (offerDetails, ipOffersByRegion, countryList) => {
+    return offerDetails
+      .filter(({ productRegion }) => ipOffersByRegion.includes(productRegion))
+      .map((ipOffer) => {
+        set(
+          ipOffer,
+          'productDisplayName',
+          `${ipOffer.productShortName} - ${ipOffer.price.text}`,
+        );
+        set(
+          ipOffer,
+          'countries',
+          ipOffer.countries.filter(
+            ({ code }) => countryList.indexOf(code.toLowerCase()) > -1,
+          ),
+        );
+        return ipOffer;
```

Comment:
You could do the following to avoid using `set`:
```
return {...ipOffer, productDisplayName: '', countries: []...}
```

Reply:
Taken care as per your suggestion